app/routes/upload.step1.tsx

```diff
@@ -0,0 +1,121 @@
+import { ActionFunctionArgs } from '@remix-run/node'
+import { Form, redirect, useNavigation, useRouteError } from '@remix-run/react'
+import { useState } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+import { Button } from '~/components/ui/button'
+import { cn } from '~/lib/utils'
+import { supabase } from '~/supabaseClient'
+
+// Action to handle form submission
+export const action = async ({ request }: ActionFunctionArgs) => {
+  const body = await request.formData()
+  const file = body.get('cv-file') as File | null
+  if (!file) throw Error('Could not get file.')
+
+  const uniqueFileName = `${uuidv4()}-${file.name}`
+  const { data, error } = await supabase.storage
+    .from('resumess')
```

Comment:
**issue**: Security concern: Ensure file type and size are validated on the server-side as well.